arch/arm/core/cortex_m/Kconfig

```diff
@@ -182,4 +190,12 @@ config ZERO_LATENCY_IRQS
 	  Note that this is a somewhat dangerous option: ISRs of priority zero
 	  interrupts cannot use any kernel functionality.
 
+config IRQ_SOFT_RELAY
+	bool
+	prompt "Enable Cortex-M0 Vector Table soft relay"
+	default y if CPU_CORTEX_M0 && !CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP
```

Comment:
Thinking about this, let's do the following for now:
```
default n
depends on CPU_CORTEX_M0 && !CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP
```
I'm working on some approaches into how to enable this automatically for MCUboot, but for now let's not force it for everybody.